### ISSUE TITLE
aar doesnt pull in required dependencies

### ISSUE THREAD

**Issue by georgepapas** [patrickfav/armadillo#2]
armadillo depends on bytes and hkdf libraries. It would be nice if there were specified as transitive dependencies in the pom and pulled in automatically.
```
  implementation "at.favre.lib:bytes:0.4.6"
  implementation "at.favre.lib:hkdf:1.0.0"
```

Cheers

**Comment by georgepapas** [comment]
also required

``` org.mindrot:jbcrypt:0.4  ```

**Comment by SAGARSURI** [comment]
`implementation 'com.jakewharton.timber:timber:4.6.0'`

**Comment by patrickfav** [comment]
This is very strange. I guess the new Gradle syntax "compile" vs "implementation" breaks adding the dependencies to the deployed pom file, I will look into it.

**Comment by patrickfav** [comment]
I've released [v0.4.0](https://github.com/patrickfav/armadillo/releases/tag/v0.4.0), which should include all relevant dependencies. I haven't tryed it yet.

**Comment by Javinator9889** [comment]
![Continues failing](https://user-images.githubusercontent.com/25952165/39008231-5b6229ce-4408-11e8-95da-5a474e3ce1c4.png)

I have just added the new lib and with both compile and/or implementation, it crashes (seems like the file is not present in the repo). The mentioned URL of the pic: https://jcenter.bintray.com/at/favre/lib/hkdf/1.0.0/hkdf-1.0.0.aar

**Comment by patrickfav** [comment]
Strange that it gets an *.aar, since this is a *.jar, will look into it. Looks like the maven plugin messes the dependency types up.

**Comment by patrickfav** [comment]
So this should finally be resolved with 0.4.1, tested it with a simple app. Closing the ticket, if there are further problems, feel free to reopen or make another report. Cheers.
